1.9/metrics/datadog.md

```diff
@@ -0,0 +1,135 @@
+As a stopgap during testing, you may be able to manually run the Datadog plugin on your agents by running it as a Marathon task. You must first upload your binary to a web server that's visible to your cluster, then create a Marathon application like the following (with customized `cmd`, `instances`, and `uris` to meet your needs):
+
+```json
+{
+  "cmd": "chmod +x ./dcos-metrics-* && ./dcos-metrics-* -dcos-role agent -auth-token <CONTENT OF 'dcos config show core.dcos_acs_token'>",
+  "instances": NUMBER_OF_AGENTS,
+  "uris": [ "https://YOURFILEHOST.COM/dcos-metrics-datadog_plugin-YOURBUILDVERSION" ],
+  "id": "test-datadog-plugin",
+  "cpus": 0.1,
+  "mem": 128,
+  "disk": 0,
+  "acceptedResourceRoles": [ "slave_public", "*" ]
+}
+```
+
+### Install the DC/OS Datadog metrics plugin
+When you're happy with the test results, you'll need to install the plugin into your cluster. For each host in your cluster, transfer your binary for the plugin and then add a systemd unit to manage the service. This unit differs slightly between agent and master hosts.
+
+### Create a Valid Auth Token for DC/OS
+Follow the instructions based on whether you are using Enterprise or open source DC/OS:
+
+- [Enterprise DC/OS](https://docs.mesosphere.com/1.9/security/service-auth/custom-service-auth/)
+- [Open source DC/OS](https://dcos.io/docs/administration/id-and-access-mgt/managing-authentication/) 
+
+You will use this auth token below.
+
+### Deploy the Metrics Plugin to Every Cluster Host
+
+1.  Copy the Datadog plugin from a local host to your remote host (`my.host:/usr/bin`):
+
+    ```bash
+    scp dcos-metrics-datadog-plugin-1.0.0-rc7 my.host:/usr/bin
+    ```
+    
+    
+1.  [SSH to your master node] and assign permission to the plugin. 
+
+    ```bash
+    dcos node ssh --master-proxy --leader
+    chmod 0755 /usr/bin/dcos-metrics-datadog-plugin-1.0.0-rc7
+    ```
+
+### Master Systemd Unit
+Create a master systemd unit file on your master node and save as `/etc/systemd/system/dcos-metrics-datadog-plugin.service`. 
+
+```
+[Unit]
+Description=DC/OS Datadog Metrics Plugin (master)
+
+[Service]
+ExecStart=/usr/bin/dcos-metrics-datadog-plugin-1.0.0rc7 -dcos-role master -metrics-port 80 -auth-token <MY_AUTH_TOKEN>
+```
+
+### Agent Systemd Unit
+Add a agent systemd unit file on your master node and save as `/etc/systemd/system/dcos-metrics-datadog-plugin.service`.
+
+```
+[Unit]
+Description=DC/OS Datadog Metrics Plugin (agent)
+
+[Service]
+ExecStart=/usr/bin/dcos-metrics-datadog-plugin-1.0.0rc7 -dcos-role agent -auth-token <MY_AUTH_TOKEN>
+```
+
+*Note:* This plugin runs on the agent Admin Router port `:61001` by default, so the port is not passed as it was in the master version of the service.
+
+### Enable, Start, and Verify
+
+1.  Enable and start the Datadog plugin.
+
+    ```bash
+    systemctl enable dcos-metrics-datadog-plugin && systemctl start dcos-metrics-datadog-plugin
+    ```
+    
+1.  View the system logs and verify that everything is okay.
```

Comment:
There should be no failure to start - ie the service is actively producing logs which are not failures from systemd. Additionally, there should be no error-level logs. Here's an example of error-level log:
```
May 05 09:03:57 ip-10-0-2-9.us-west-2.compute.internal dcos-metrics[32138]: time="2017-05-05T09:03:56Z" level=error msg="json: unsupported value: NaN" collector=framework
```